tests/Functional/Schema/PostgreSQLSchemaManagerTest.php

```diff
@@ -561,6 +564,73 @@ public function testAlterTableAutoIncrementIntToBigInt(
         self::assertTrue($tableFinal->getColumn('id')->getAutoincrement());
     }
 
+    public function testListTableColumnsOidConflictWithNonTableObject(): void
+    {
+        $semanticVersion = $this->connection->fetchOne('SHOW SERVER_VERSION');
+        if ($semanticVersion === false) {
+            throw new Exception("Query to fetch server version failed when it shouldn't have.");
+        }
+
+        $majorVersion = explode('.', $semanticVersion)[0];
+
+        if ($majorVersion === '9') {
+            self::markTestSkipped('Manually setting the Oid is not supported in 9.4');
+        }
+
+        $table = 'test_list_table_columns_oid_conflicts';
+        $this->connection->executeStatement(sprintf('CREATE TABLE IF NOT EXISTS %s(id INT NOT NULL)', $table));
+        $beforeColumns = $this->schemaManager->listTableColumns($table);
+        $this->assertArrayHasKey('id', $beforeColumns);
+
+        $this->connection->executeStatement('CREATE EXTENSION IF NOT EXISTS pg_prewarm');
+        $originalTableOid = $this->connection->fetchOne(
+            'SELECT oid FROM pg_class WHERE pg_class.relname = ?',
+            [$table],
+        );
+
+        $getConflictingOidSql = <<<'SQL'
+SELECT objid
+FROM pg_depend
+    JOIN pg_extension as ex on ex.oid = pg_depend.refobjid
+    WHERE ex.extname = 'pg_prewarm'
+    ORDER BY objid
+    LIMIT 1
+SQL;
+        $conflictingOid       = $this->connection->fetchOne($getConflictingOidSql);
+
+        $this->connection->executeStatement(
+            'UPDATE pg_attribute SET attrelid = ? WHERE attrelid = ?',
+            [$conflictingOid, $originalTableOid],
+        );
+        $this->connection->executeStatement(
+            'UPDATE pg_description SET objoid = ? WHERE objoid = ?',
+            [$conflictingOid, $originalTableOid],
+        );
+        $this->connection->executeStatement(
+            'UPDATE pg_class SET oid = ? WHERE oid = ?',
+            [$conflictingOid, $originalTableOid],
+        );
+
+        $afterColumns = $this->schemaManager->listTableColumns($table);
+        $this->assertArrayHasKey('id', $afterColumns);
+
+        // revert to the original oid
+        $this->connection->executeStatement(
+            'UPDATE pg_attribute SET attrelid = ? WHERE attrelid = ?',
+            [$originalTableOid, $conflictingOid],
+        );
+        $this->connection->executeStatement(
+            'UPDATE pg_description SET objoid = ? WHERE objoid = ?',
+            [$originalTableOid, $conflictingOid],
+        );
+        $this->connection->executeStatement(
+            'UPDATE pg_class SET oid = ? WHERE oid = ?',
+            [$originalTableOid, $conflictingOid],
+        );
+
+        $this->connection->executeStatement(sprintf('DROP TABLE IF EXISTS %s', $table));
```

Comment:
Why do we need to change thing in the database _after_ we have made our assertion?

Reply:
Since I'm modifying the pg catalog directly, I wanted to put the original values back so that this wouldn't affect downstream tests.  I tried wrapping the changes in a transaction but unfortunately that didn't work and pg threw an error when rolling back.  Even if it doesn't cause problems now, if it does affect a test in the future I think it will be impossible to debug for whatever poor soul runs into it.

Reply:
Okay, but that means, if the assertion fails, your cleanup won't run?

Reply:
I moved the assertion after undoing the database changes to account for this.